sources/ldap: set password_change_date in check_pwd_last_set to avoid loop ending user sessions

tograss opened the pull request:
<!--
👋 Hi there! Welcome.

  Please check the Contributing guidelines: https://docs.goauthentik.io/docs/developer-docs/#how-can-i-contribute
-->

## Details

<!--
Explain what this PR changes, what the rationale behind the change is, if any new requirements are introduced or any breaking changes caused by this PR.

Ideally also link an Issue for context that this PR will close using `closes #`
-->
closes #11436
**Explanation:**
In (ms_)check_pwd_last_set the line 
` if created or pwd_last_set > user.password_change_date:`
test if the pw in the ldap store is newer than the authentik one. If so
`
user.set_unusable_password()
user.save()
`
Is executed. Changing the  user password (correctly) ends all user sessions. See https://docs.djangoproject.com/en/5.1/topics/auth/default/#:~:text=Django%20also%20provides%20views%20and,log%20out%20all%20their%20sessions.
However, since user.password_change_date is not set the next ldap_sync will lockout the user again.
I think this wasn't a problem when during login the user pw was always saved to the authentik database. Since this changed a while ago this is now a bug.

The fix is to simply set the password_change_date to the ldap date. One might argue it should be the time the ldap sync run, but I think it is easier to understand for an administrator if the ldap date and the authentik date are the same.

---

## Checklist

-   [x] Local tests pass (`ak test authentik/`)
-   [x] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

tograss committed:
Set passord_change_date in check_pwd_last_set

tograss committed:
Merge branch 'goauthentik:main' into main

BeryJu commented on authentik/sources/ldap/sync/vendor/freeipa.py:
```diff
@@ -38,6 +38,7 @@ def check_pwd_last_set(self, attributes: dict[str, Any], user: User, created: bo
                 pwd_last_set=pwd_last_set,
             )
             user.set_unusable_password()
+            user.password_change_date = pwd_last_set
```
the more correct solution for this would be something like

```diff
diff --git a/authentik/core/models.py b/authentik/core/models.py
index 85e8901ed..470436db0 100644
--- a/authentik/core/models.py
+++ b/authentik/core/models.py
@@ -340,6 +340,10 @@ class User(SerializerModel, GuardianUserMixin, AttributesMixin, AbstractUser):
         self.password_change_date = now()
         return super().set_password(raw_password)

+    def set_unusable_password(self):
+        self.password_change_date = now()
+        return super().set_unusable_password()
+
     def check_password(self, raw_password: str) -> bool:
         """
         Return a boolean of whether the raw_password was correct. Handles
```

instead of setting the date everywhere

tograss replied:
I added a combination of a) and c)

tograss committed:
Refactor freeipa.py and ms_ad.py to use a common check_pwd_last_set method from the base class

tograss committed:
Override set_unusable_password in user model.
Our version always updates the user.password_change_date either to now or a given value

tograss committed:
lint

tograss committed:
Ensure that during user create, the changed_password_date does not change after the event is logged

tograss commented:
@BeryJu 
Hi Bery, can you provide an estimate when you can review / merge this pull request. The related issues will effect more and more users as time progresses.

tograss commented:
Hi can I help with anything to get this fix merged ?
